File: WebShopKBS/WebShopKBS/Controllers/CustomerController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web;
using System.Web.Http;
using System.Web.SessionState;
using WebShopKBS.Models;
using WebShopKBS.Models.UserModels;
using WebShopKBS.Services;
using WebShopKBS.UoW;

namespace WebShopKBS.Controllers
{
	[RoutePrefix("customer")]
    public class CustomerController : ApiController,IRequiresSessionState
    {
	    private readonly CustomerService service;

	    public CustomerController()
	    {
		    service = new CustomerService(new UnitOfWork());
		   
		}

	    [HttpGet]
	    public IHttpActionResult Get(string username)
	    {
			var customer = service.Get(username);
		    if (customer == null)
			    return NotFound();
		    return Ok(customer);
	    }

	    [Route("order")]
	    [HttpPost]
	    public IHttpActionResult GetBill([FromBody] Order order)
	    {
		    var bonusCreditsSpent = 0;
		    var bill = service.GetBill(order, bonusCreditsSpent);
		    if (bill == null)
			    return BadRequest("Something went wrong with forming your bill, please try again later");
		    return Ok(bill);
	    }

	    [Route("confirm")]
	    [HttpPost]
	    public IHttpActionResult PlaceOrder(Order order)
	    {
		    var completedOrder = service.PlaceOrder(order);
		    if (completedOrder == null)
			    return BadRequest("Something went wrong with placing your order, please try again later");
		    return Ok(completedOrder);
	    }

	    [Route("cart")]
	    public IHttpActionResult AddToCart(Item item)
	    {
		    if (HttpContext.Current.Session["cart"] == null)
			    HttpContext.Current.Session["cart"] = new BillDTO();

			(HttpContext.Current.Session["cart"] as BillDTO).Items.Add(item);

			return Ok(HttpContext.Current.Session["cart"]);
	    }

	    [Route("cart")]
		[HttpGet]
	    public IHttpActionResult GetCart()
	    {
		    return Ok(HttpContext.Current.Session["cart"]);
	    }

	    [Route("removeFromCart")]
	    [HttpPost]
	    public IHttpActionResult RemoveFromCart(Item item)
	    {
		    var cart = (HttpContext.Current.Session["cart"] as BillDTO);
		    Item itemToRemove = null;
		    foreach (var cartItem in cart.Items)
		    {
			    if (item.Name.Equals(cartItem.Name))
				    itemToRemove = cartItem;
		    }
		    cart.Items.Remove(itemToRemove);
		    HttpContext.Current.Session["cart"] = cart;
			return Ok(HttpContext.Current.Session["cart"]);

	    }
 	}
}

File: WebShopKBS/WebShopKBS/App/Controllers/CustomerController.js
angular.module('customer.controllers', [])
.controller('customerController',
		['$scope','$rootScope', '$window', '$state', 'customerService',
			function ($scope, $rootScope, $window, $state, customerService) {
				console.log('customer controller');

				$scope.getProfile = function (username) {
					customerService.getProfile(username).then(function (response) {
						$scope.profile = response.data;
					});
				}

				$scope.getBill = function (order) {
					customerService.getBill(order).then(function (response) {
						$scope.bill = response.data;
					});
				}

				$scope.placeOrder = function (order) {
					customerService.placeOrder(order).then(function (response) {
						console.log(response.data);
					});
				}

				$scope.removeFromCart = function(item) {
					var index = $scope.cart.items.indexOf(item);
					$scope.cart.items.splice(index, 1);
					customerService.removeFromCart(item).then(function(response) {
						$scope.cart = response.data;
						console.log(response.data);
					});
				}

				$scope.getBill = function (creditsSpent) {
					console.log('check');
					var order = {};
					order.customerId = $rootScope.user.username;
					order.items = [];
					order.bill = 0;
					for (var i = 0; i < $scope.cart.items.length; i++) {
						var orderItem = {};
						orderItem.ItemId = $scope.cart.items[i].id;
						orderItem.index = i;
						orderItem.item = $scope.cart.items[i];
						orderItem.price = $scope.cart.items[i].price;
						orderItem.count = $scope.cart.items[i].count;
						orderItem.totalPrice = $scope.cart.items[i].price * $scope.cart.items[i].count;
						order.items.push(orderItem);
						order.bill += orderItem.totalPrice;
					}
					customerService.getBill(order, creditsSpent).then(function(response) {
						$rootScope.finalBill = response.data;
						$state.go('customer.bill');
					});
				}

				var getContent = function () {
					customerService.getCart().then(function (response) {
						$scope.cart = response.data;
						console.log(response.data);
					});
				}
				getContent();


			}])